### PR TITLE
89% test coverage and several fixed bugs

### PR DESCRIPTION
**Pull request by paulgrove**
Hi Steve,

Hope you are well!

We found a bug in syslog-client, so I decided to write some tests and fix that bug and others I found while writing the tests.

The main issue I was addressing was that `this.transport_` was never being reused.  Subsequently bugs relating to the propagation of errors, and parameter checking were found and identified while writing tests.

I've updated the change log, and bumped the version in package.json.

Let me know if you want me to make any changes to the pull request.

If your in town, give me a call!

Regards,
Paul

### CODE REVIEW

**Comment by paulgrove** [comment]
When you get to this, I was wondering how you felt about the dependencies.

Currently the dependencies are only required for running the tests, Istanbul is already added as a dev-dependency.  I realise that npm doesn't run the tests on install, so we could move all the dependencies I added to the dev-dependencies.

Additionally I removed bluebird as a dependency, I used Promises in the tests, however they are not used in the module, and the tests will run using the native Promise.

Of course now bluebird has been removed, the tests will not run on older versions of nodejs which do not support native Promises.

tldr;
We could move all the dependencies to the dev-dependencies and re-include bluebird (to dev-deps) so that the tests will run on older nodes. thoughts?

**Comment by MarkHerhold** [comment]
Nice! Personally, I'd prefer the removal of Bluebird, besides users can always polyfill it, e.g. `global.Promise = require('bluebird')`

**Comment by paulgrove** [comment]
I agree, and have already removed Bluebird, and to clarify I've only used Promises in the tests, Promises are not needed to use this fork.

Another question I have is if I should move the testing libraries (mocha, chai) into dev dependencies or not?  Does NPM ever run the tests on install? otherwise I see no problem in moving them, reducing the dependencies on install back to zero.

**Comment by MarkHerhold** [comment]
I'm certaintly not the one calling the shots on this repository, but every project I have ever seen and written myself has the test dependencies classified as dev dependencies. NPM will not run tests on install unless the `preinstall` or `postinstall` script hooks in `package.json` are defined to run the tests, which is not the case for this repo.

**Comment by MarkHerhold** [comment]
@stephenwvickers Any thoughts on pulling this in?

**Comment by paulgrove** [comment]
@MarkHerhold I believe @stephenwvickers is planning on transferring this project to me, I'll merge this change at that time.

**Comment by MarkHerhold** [comment]
@paulgrove SGTM! Thanks for working this out both of you.

**Comment by stephenwvickers** [comment]
@MarkHerhold no problem!

For context, I worked with @paulgrove previously, which is why I originally authored the module.

Since I no longer have a vested interest in it and he does I thought it made sense to hand over control to him.

You could probably tell that it is no longer near the top of my priority list :S

Steve
